Build error: "use of deleted function"

Hello,
Thank you for writing this async_mqtt, it looks great!
I'm trying to build version 2.0.0 with GCC 9.3.0. I'm cross-compiling using arm-oe-linux-gnueabi-g++.

The build is failing right from the start:

`
/arm-oe-linux-gnueabi-g++ --sysroot=/home/bjc/quectel_sensor_sdk/rm520nglaar/ql-ol-extsdk/ql-sysroots -DASYNC_MQTT_USE_TLS -Iasync_mqtt-2.0.0/include -isystem /home/bjc/quectel_sensor_sdk/boost_1_81_0 -march=armv7-a -marm -mfpu=neon -mfloat-abi=hard -Os -DNDEBUG -std=c++2a -MD -MT example/CMakeFiles/ep_cb_mqtt_client.dir/ep_cb_mqtt_client.cpp.o -MF CMakeFiles/ep_cb_mqtt_client.dir/ep_cb_mqtt_client.cpp.o.d -o CMakeFiles/ep_cb_mqtt_client.dir/ep_cb_mqtt_client.cpp.o -c async_mqtt-2.0.0/example/ep_cb_mqtt_client.cpp
In file included from async_mqtt-2.0.0/include/async_mqtt/packet/property_variant.hpp:11,
                 from async_mqtt-2.0.0/include/async_mqtt/packet/will.hpp:12,
                 from async_mqtt-2.0.0/include/async_mqtt/packet/v3_1_1_connect.hpp:26,
                 from async_mqtt-2.0.0/include/async_mqtt/packet/packet_variant.hpp:11,
                 from async_mqtt-2.0.0/include/async_mqtt/buffer_to_packet_variant.hpp:12,
                 from async_mqtt-2.0.0/include/async_mqtt/all.hpp:11,
                 from async_mqtt-2.0.0/example/ep_cb_mqtt_client.cpp:12:
async_mqtt-2.0.0/include/async_mqtt/packet/property.hpp: In constructor ‘async_mqtt::property::detail::binary_property::binary_property(async_mqtt::property::id, async_mqtt::buffer)’:
async_mqtt-2.0.0/include/async_mqtt/packet/property.hpp:142:19: error: use of deleted function ‘async_mqtt::buffer::buffer(const async_mqtt::buffer&)’
  142 |          length_(2) // size 2
      |                   ^
In file included from async_mqtt-2.0.0/include/async_mqtt/all.hpp:10,
                 from async_mqtt-2.0.0/example/ep_cb_mqtt_client.cpp:12:
async_mqtt-2.0.0/include/async_mqtt/buffer.hpp:48:5: note: ‘async_mqtt::buffer::buffer(const async_mqtt::buffer&) noexcept’ is implicitly deleted because its exception-specification does not match the implicit exception-specification ‘’
   48 |     buffer(buffer const& other) noexcept = default;
      |     ^~~~~~
`

--- a/.github/workflows/gha.yml
+++ b/.github/workflows/gha.yml
@@ -82,6 +82,9 @@ jobs:
       with:
         path: usr
         key: ${{ runner.os }}-boost-20230507
+    - name: install clang++15
+      run: |
+         sudo apt-get install clang-15 clang++-15
     - name: Build boost
       if: steps.cache-boost.outputs.cache-hit != 'true'
       run: ./.github/depends/boost.sh -b 64 -t gcc
@@ -101,18 +104,13 @@ jobs:
         [ ${{ matrix.pattern }} == 2 ] && \
         export CFLAGS=${NS_CFLAGS} && export CXXFLAGS=${NS_CXXFLAGS} && export LDFLAGS=${NS_LDFLAGS}
         [ ${{ matrix.pattern }} == 3 ] && \
-        gcc --version \
-        export CFLAGS=${NS_CFLAGS} && export CXXFLAGS="${NS_CXXFLAGS_GCC} "&& export LDFLAGS=${NS_LDFLAGS}
-        [ ${{ matrix.pattern }} == 0 ] && FLAGS="-DCMAKE_CXX_COMPILER=clang++ -DASYNC_MQTT_USE_TLS=OFF -DASYNC_MQTT_USE_WS=OFF -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=ON "
-        [ ${{ matrix.pattern }} == 1 ] && FLAGS="-DCMAKE_CXX_COMPILER=clang++ -DASYNC_MQTT_USE_TLS=ON  -DASYNC_MQTT_USE_WS=OFF -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=OFF"
-        [ ${{ matrix.pattern }} == 2 ] && FLAGS="-DCMAKE_CXX_COMPILER=clang++ -DASYNC_MQTT_USE_TLS=OFF -DASYNC_MQTT_USE_WS=ON  -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=OFF"
-        [ ${{ matrix.pattern }} == 3 ] && FLAGS="-DCMAKE_CXX_COMPILER=g++-12  -DASYNC_MQTT_USE_TLS=ON  -DASYNC_MQTT_USE_WS=ON  -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=OFF"
+        export CFLAGS=${NS_CFLAGS} && export CXXFLAGS=${NS_CXXFLAGS_GCC} && export LDFLAGS=${NS_LDFLAGS}
+        [ ${{ matrix.pattern }} == 0 ] && FLAGS="-DCMAKE_CXX_COMPILER=clang++-15 -DASYNC_MQTT_USE_TLS=OFF -DASYNC_MQTT_USE_WS=OFF -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=ON "
+        [ ${{ matrix.pattern }} == 1 ] && FLAGS="-DCMAKE_CXX_COMPILER=clang++-15 -DASYNC_MQTT_USE_TLS=ON  -DASYNC_MQTT_USE_WS=OFF -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=OFF"
+        [ ${{ matrix.pattern }} == 2 ] && FLAGS="-DCMAKE_CXX_COMPILER=clang++-15 -DASYNC_MQTT_USE_TLS=OFF -DASYNC_MQTT_USE_WS=ON  -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=OFF"
+        [ ${{ matrix.pattern }} == 3 ] && FLAGS="-DCMAKE_CXX_COMPILER=g++-12     -DASYNC_MQTT_USE_TLS=ON  -DASYNC_MQTT_USE_WS=ON  -DASYNC_MQTT_USE_STR_CHECK=ON  -DASYNC_MQTT_USE_LOG=OFF"
         FLAGS="$FLAGS -DASYNC_MQTT_BUILD_TOOLS=ON -DASYNC_MQTT_BUILD_EXAMPLES=ON -DASYNC_MQTT_BUILD_UNIT_TESTS=ON -DASYNC_MQTT_BUILD_SYSTEM_TESTS=ON"
-         echo "begin"
-         echo  ${{env.BOOST_ROOT}}
-         echo "end"
-         ls -al /home/runner/work/async_mqtt/async_mqtt/usr
-         BOOST_ROOT=/home/runner/work/async_mqtt/async_mqtt/usr cmake -S ${{ github.workspace }} -B ${{ runner.temp }} ${FLAGS} -DCMAKE_C_FLAGS="${CFLAGS}" -DCMAKE_CXX_FLAGS="${CXXFLAGS}" -DCMAKE_EXE_LINKER_FLAGS="${LDFLAGS}"
+        BOOST_ROOT=/home/runner/work/async_mqtt/async_mqtt/usr cmake -S ${{ github.workspace }} -B ${{ runner.temp }} ${FLAGS} -DCMAKE_C_FLAGS="${CFLAGS}" -DCMAKE_CXX_FLAGS="${CXXFLAGS}" -DCMAKE_EXE_LINKER_FLAGS="${LDFLAGS}"
     - name: Check Header Dependencies
       run: |
         cmake --build ${{ runner.temp }} --parallel $(nproc) --clean-first --target check_deps

--- a/include/async_mqtt/buffer.hpp
+++ b/include/async_mqtt/buffer.hpp
@@ -45,7 +45,6 @@ public:
     static constexpr size_type npos = string_view::npos;
 
     constexpr buffer() noexcept = default;
-    buffer(buffer const& other) noexcept = default;
     constexpr buffer(char const* s, std::size_t count)
         :view_{s, count}
     {}
@@ -116,8 +115,6 @@ public:
         : view_{&*first, static_cast<std::size_t>(std::distance(first, last))},
           life_{force_move(life)}
     {}
-
-    buffer& operator=(buffer const& buf) noexcept = default;
 
     constexpr const_iterator begin() const noexcept {
         return view_.begin();

